overlord/servicestate/internal/quotas.go

```diff
@@ -103,13 +103,22 @@ func PatchQuotas(st *state.State, grps ...*quota.Group) (map[string]*quota.Group
 		return nil, fmt.Errorf("cannot update quota%s %s: %v", plural, updated, err)
 	}
 
+	// Verify that the update of the new quota groups will result in
+	// correct nesting of groups. Execute this verification after updating
+	// group pointers in ResolveCrossReferences.
+	for _, grp := range grps {
+		if err := grp.VerifyNesting(); err != nil {
+			return nil, fmt.Errorf("cannot update quota %q: %v", grp.Name, err)
```

Comment:
do we have a test that triggers this?

Reply:
Yes, we have a test (TestQuotaSnapCorrectlyDetectErrorsIfCreatingSubgroupsFirst) that triggers these errors, and the spread test triggers them.

Reply:
Added additional (dedicated) unit tests for this

Reply:
Added a case in same package that triggers this error